docs/source/examples/performance-training.rst

```diff
@@ -0,0 +1,408 @@
+- `TensorFlow Distribution Strategies <https://github.com/skypilot-org/skypilot/tree/master/examples/tensorflow>`_
+
+The choice of framework depends on your specific needs, but all can be easily configured through SkyPilot's YAML specification.
+
+Best practices
+--------------
+
+High performance instances
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Choose high performance instances for optimal training performance. SkyPilot allows you to specify instance types with powerful GPUs and high-bandwidth networking:
+
+- Use the latest GPU accelerators (A100, H100, etc.) for faster training
+- Select instances with NVLink for improved multi-GPU communication
+- Consider instances with higher memory bandwidth and higher device memory for large models
+
+Example configuration:
+
+.. code-block:: yaml
+
+  resources:
+    accelerators: 
+      A100:1  
+      A100-80GB:1  
+      H100:1  
+
+Use best disk_tier
+~~~~~~~~~~~~~~~~~~~~~~~
+Fast storage is critical for loading and storing data and model checkpoints.
+SkyPilot's ``disk_tier`` option supports the fastest available storage with high-performance local SSDs to reduce I/O bottlenecks.
+
+Example configuration:
+
+.. code-block:: yaml
+
+  resources:
+    disk_tier: best  # Use highest performance disk tier.
+
+
+MOUNT_CACHED for checkpointing
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+MOUNT_CACHED provides high performance writing, making it ideal for model checkpoints, logs, and other outputs with fast local writes. Unlike MOUNT mode, it supports all write operations without limitations on file operations. It offers fast file access through a local VFS (Virtual File System) cache implemented by `rclone <https://rclone.org/>`__ that provides near-local disk performance. 
+
+Example configuration:
+
+.. code-block:: yaml
+
+    file_mounts:
+      /checkpoint:
+        name: my-checkpoint-bucket  
+        mode: MOUNT_CACHED
+
+.. note::
+   
+   In MOUNT_CACHED mode, files only begin uploading after they are closed by all processes. By default, SkyPilot uses sequential transfers to maintain file order, which can cause the cache to grow if files are written faster than they can be uploaded.
+
+   When multiple jobs share a node (e.g., using fractional GPUs), be aware of these important considerations:
+   
+   - A job is only marked complete after all files in the cache are flushed to the remote bucket, including those from other jobs.
+   - The total disk space required equals the sum of all jobs' cached data.
+   - Jobs that finish quickly might need to wait for other jobs' files to be uploaded.
+   - Ensure the disk tier is fast enough to handle the combined write load of all jobs.
+
+To see the difference between MOUNT and MOUNT_CACHED, see :ref:`storage mounting modes <storage-mounting-modes>`.
+
+Robust checkpointing for spot instances
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When using spot instances, robust checkpointing is crucial for recovering from preemptions. 
```

Comment:
There is an assumption here: "you should design your job to try and load checkpoints when it starts, because there's no way to know if it's the first run or a restart"
It might be good to clarify the high level thing at the beginning. Specifically:
1. Write checkpoints periodically during iteration code.
2. When you first start, try to load checkpoints. On the first run, this will do nothing, but on restarts it will pick things up where they left off.

My sense is that 2. is slightly unintuitive. If you disagree, maybe it's fine.